source/extended-json/extended-json-query.md

```diff
@@ -0,0 +1,56 @@
+# Extended JSON Query
+
+## Abstract
+
+Extended JSON query variation is a superset of Extended JSON. It is
```

Comment:
This text seems to be describing Shell BSON syntax, which is neither a JSON format nor a superset of Extended JSON

Reply:
@samontea can you help shed some light on this?

Is the Shell BSON syntax documented @addaleax?

Reply:
Yeah, I don't describing it as a superset of Extended JSON is accurate. The proposed format wouldn't accept, for example, `{$numberInt: <number as string>}` would it? See this table in the extended JSON spec: https://github.com/mongodb/specifications/blob/master/source/extended-json/extended-json.md#conversion-table.

Also calling it "Extended JSON Query" seems like an inaccuate name. I thought we settled on a name like "BSON Text Format", since the folks from [SPM-3714](https://jira.mongodb.org/browse/SPM-3714) argue (reasonably, I think) that a non-JSON format is more readable.

Reply:
> Is the Shell BSON syntax documented @addaleax?

https://www.mongodb.com/docs/mongodb-shell/reference/data-types/ documents some of the constructors provided by the shell, but it's not exhaustive.

> I thought we settled on a name like "BSON Text Format", since the folks from [SPM-3714](https://jira.mongodb.org/browse/SPM-3714) argue (reasonably, I think) that a non-JSON format is more readable.

You can call it whatever you want, but if you want a name that's been "settled on", we've put in a lot of effort on our team to standardize on using "Shell BSON syntax" internally (to distinguish it from things like Extended JSON), and that long predates SPM-3714 _and_ is being used by the team actually owning mongosh.

Reply:
@dstorch the proposed format does support `{$numberInt: <number as string>}`. i tested that one explicitly but a quick pass over the code confirms the rest are supported. the method which parses this is called `fromFuzzerJson()`. but i don't think "fuzzer JSON" should be a syntax.

afaiu, conveniently reusing the fuzzer generated queries was prefered both to reduce effort and have queries with the BSON syntax (incl date constructor and native regex). I found [this documentation of tojsononeline()](https://www.mongodb.com/docs/mongodb-shell/reference/ejson/stringify/#legacy-tojsononeline--) but it's not very helpful. If tojsononeline() is documented, we can use that to specify the input syntax to QueryTester (even if the implementation accepts a more general grammar).